### PR TITLE
include: inttypes: add SCN defines

### PR DESCRIPTION
**Pull request by damianloew**
## Description

 - include: inttypes: add SCN defines
 
## Motivation and Context

JIRA: PD-272

`SCNU64` macro needed in mbedtls incoming port.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (using SCNu, when building mbedtls).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: include/arch/armv7/stdint.h [hunk 1]
```diff
@@ -146,5 +146,6 @@ typedef uint64_t uintmax_t;
 #define _PRI_16 ""
 #define _PRI_32 ""
 #define _PRI_64 "ll"
+#define _PRI_PTR "ll"
```
Comment (nalajcie) [inline include/arch/armv7/stdint.h:149]: IMHO this target has 32-bit pointers (see `UINTPTR_MAX` for example) so the correct prefix should be `""`. Applies for other 32-bit arch also.

**Review by nalajcie — CHANGES_REQUESTED** [review]

File: include/arch/armv7/stdint.h (hunk 1 above)
Comment (damianloew) [reply include/arch/armv7/stdint.h]: I wasn't sure and wanted someone to verify if it's correct. Thanks, changed.

**Review by nalajcie — APPROVED** [review]

**Review by anglov — CHANGES_REQUESTED** [review]
This comment is for whole code.

1. Please sort and group defines in sane order. – If you prefer yours, then please reorder current one.
2. To do it correctly you need to use platform-specific prefixes. 
eg. you need additional prefix like _PRI_LEAST(8/16/32/64) – as size of those types is architecture-defined  
3. As you use _PRI in SCN macros, then maybe it should be renamed eg. to _FMT and use valid prefixes for both printf and scanf – currently those print prefixes causes implicit cast to higher type. Hope that nothing will break with that change.

**Comment by damianloew** [comment]
> This comment is for whole code.
> 
> 1. Please sort and group defines in sane order. – If you prefer yours, then please reorder current one.
> 2. To do it correctly you need to use platform-specific prefixes.
>    eg. you need additional prefix like _PRI_LEAST(8/16/32/64) – as size of those types is architecture-defined
> 3. As you use _PRI in SCN macros, then maybe it should be renamed eg. to _FMT and use valid prefixes for both printf and scanf – currently those print prefixes causes implicit cast to higher type. Hope that nothing will break with that change.

Thanks, I've adopted your suggestions. Please check last force push.

File: include/arch/riscv64/stdint.h [hunk 2]
```diff
@@ -146,5 +146,6 @@ typedef uint64_t uintmax_t;
 #define _PRI_16 ""
 #define _PRI_32 ""
 #define _PRI_64 "l"
+#define _PRI_PTR "l"
```
[the remark this answers is not in the document]
Comment (damianloew) [reply include/arch/riscv64/stdint.h:150]: Thanks, corrected.

[the remark this answers is not in the document]
Comment (anglov) [reply include/arch/riscv64/stdint.h:150]: Still missing defines – now _SCN_X

**Review by anglov — CHANGES_REQUESTED** [review]

File: include/arch/riscv64/stdint.h [hunk 3]
```diff
@@ -146,5 +146,19 @@ typedef uint64_t uintmax_t;
 #define _PRI_16 ""
 #define _PRI_32 ""
 #define _PRI_64 "l"
+#define _SCN_8   "hh"
+#define _SCN_16  "h"
+#define _SCN_32  ""
+#define _SCN_64  "ll"
```
Comment (anglov) [inline include/arch/riscv64/stdint.h:152]: We use (unsigned) long for (u)int64_t type for this platform so you should use l prefix. Also compatible with _PRI_64

**Review by anglov — CHANGES_REQUESTED** [review]

File: include/arch/riscv64/stdint.h (hunk 3 above)
Comment (damianloew) [reply include/arch/riscv64/stdint.h]: That's right, sorry.

**Pull request merged**